server/src/service.rs

```diff
@@ -115,17 +117,41 @@ impl Service {
     #[zbus(out_args("unlocked", "prompt"))]
     pub async fn unlock(
         &mut self,
-        _objects: Vec<OwnedObjectPath>,
-    ) -> Result<(Vec<OwnedObjectPath>, ObjectPath), ServiceError> {
-        todo!()
+        objects: Vec<OwnedObjectPath>,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+    ) -> Result<(Vec<OwnedObjectPath>, OwnedObjectPath), ServiceError> {
+        let unlocked = self.set_locked(false, &objects).await?;
+        if unlocked.is_empty() {
```

Comment:
The condition here seems wrong, you would use a prompt only if there are items which don't have the expected state.

Reply:
Yeah I agree.

Reply:
Quick question: we can issue a prompt if an object doesn't have the expected state. But what should we do if all the objects (more than one) in the input parameter don't have the expected state? I tried this scenario with busctl and d-spy but, nothing happens :)

Reply:
I am not sure what you mean, the prompt should only get the items that need change that require a prompt, the remaining should be skipped from prompting.